go/private/providers.bzl

```diff
@@ -62,6 +62,7 @@ def _source_build_entry(srcs = [], deps = [], gc_goopts=[], runfiles=None, cgo_d
       cgo_exports = cgo_exports,
       cgo_archive = cgo_archive,
       want_coverage = want_coverage,
+      exclude = exclude,
```

Comment:
Consider renaming to indicate this checks the mode, and reversing the sense.

Maybe `_source_filter` below could be:

```
def _source_filter(ctx, source, mode):
  return GoSourceList(entries = [s for s in source.entries if (not s.mode_ok or s.mode_ok(ctx, mode))])
```

Reply:
It does not have to check the mode, it can exclude for any reason it likes, checking feature flags on the ctx for instance.
I picked exclude because include is confusing in a world where c++ headers exist, and filter is ambiguous

Reply:
I am going to submit this so I can keep going, but I am happy to do another PR to change the name (and invert the logic) if needed.